starfish/image/_stack.py

```diff
@@ -242,22 +276,72 @@ def show_stack(
                 out_range=numpy.float32
             ).astype(numpy.float32)
 
-        def show_plane(ax, plane, cmap="gray", title=None):
+        if p_min or p_max:
+            print("Clipping ...")
+            a_min, a_max = scoreatpercentile(linear_view, (p_min if p_min else 0, p_max if p_max else 100))
+            linear_view = numpy.clip(linear_view, a_min=a_min, a_max=a_max)
+
+        show_spot_function = self._show_spots
+
+        def show_plane(ax, plane, plane_index, cmap="gray", title=None):
             ax.imshow(plane, cmap=cmap)
+            if show_spots:
+                # this is slow. This link might have something to help:
+                # https://bastibe.de/2013-05-30-speeding-up-matplotlib.html
+                show_spot_function(show_spots.data, ax=ax, z=plane_index)
             ax.set_xticks([])
             ax.set_yticks([])
 
             if title:
                 ax.set_title(title)
 
-        @interact(plane=(0, n - 1))
-        def display_slice(plane=34):
+        @interact(plane_index=(0, n - 1))
+        def display_slice(plane_index=0):
             fig, ax = plt.subplots(figsize=figure_size)
-            show_plane(ax, linear_view[plane], title=f'{labels[plane]}', cmap=color_map)
+            show_plane(ax, linear_view[plane_index], plane_index, title=f'{labels[plane_index]}', cmap=color_map)
             plt.show()
 
         return display_slice
 
+    @staticmethod
+    def _show_spots(result_df, ax, z=None, size=1, z_dist=1.5, scale_radius=5) -> None:
+        """function to plot spot finding results on top of any image as hollow red circles
+
+        called spots are colored by category
+
+        Parameters:
+        -----------
+        img : np.ndarray
+            2-d image
```

Comment:
Can we make it a project convention to be specific about the acceptable dtypes for ndarrays?

Reply:
YES. However we'll need to go learn about what they are by testing.

Reply:
Well, this is your viz code.  Seems like you should have an idea of what you'll accept :)

Reply:
Here I can add it -- it'll take any numerical array. But other filters are way more picky.

Reply:
Really?  What if rescale=False?

Reply:
Yea, all these scaling factors are about dynamic ranges when you plot them as a colormap. 

i.e. if you have one value that's 2 billion and the rest are 0-100, you're going to get a black image. 

however, if you clip at the top at 99.5%, then you'll get a nice image where the range is 0-100.

Reply:
So if you screw things up here, you just get an ugly plot -- it'll still "work", though.